### ISSUE TITLE
Templates do not include the major version prefix

### ISSUE THREAD

**Issue by Chuxel** [devcontainers/templates#55]
To put people on the right path and avoid breaking changes, templates should reference a major version of the image.  e.g. `base:0-bullseye` not `base:bullseye`,

This was automated in vscode-dev-containers but is not here.

**Comment by samruddhikhandale** [comment]
It was intentionally changed to use `base:bullseye` instead of `base:0-bullseye` with the assumption that we've never done a breaking change within a distro. 

This is something recently discussed with @jkeech & the change was made to avoid complicating the templates.

**Comment by Chuxel** [comment]
This is one example. Python is the same right now as is Node, but dotnet, Go, C++, Anaconda, and others have `0-` prefixes. I think we want to encourage people to not use any form of "latest" here. It also hints that version specific versions are available.  What is the criteria for when we would include the prefix and when we would not?

**Comment by samruddhikhandale** [comment]
> What is the criteria for when we would include the prefix and when we would not?

I don't think we have a formalized criteria yet 🤔 

last month, this [PR](https://github.com/devcontainers/templates/pull/26) unpinned the major versions for `base` images so that each time a new major version is released, we don't have to update the Templates (eg. we needed to update with new base-ubuntu version). We discussed it was safe to do so for `base` images, as we never have/had breaking changes within a distro.

For other language images, like Python, Go, dotnet etc, I think we should have major versions (ie `0-` prefix) as it is more likely to have breaking changes. I can scan and fix if this isn't happening right now and if that's what we agree upon.
